README.md

```diff
@@ -17,37 +15,64 @@ engine and provide you with a set of assertions you can use to verify your proce
 
 ### Dependency
 
-Add the following dependency to your project
+#### Testcontainers (JDK 8+)
+
+If you are building your project with a JDK that's lower than 17 you should use this dependency. It
+starts a testcontainer in which a Zeebe test engine is running. The advantage of using this version
+instead of the embedded version is that you will be completely separated from the Java version that is
+used by the Zeebe engine. This does come at a cost. Testcontainers provide some overhead, which means
+tests will be slower. There is also the extra requirement that Docker must be running to execute the tests.
 
 ```xml
 <dependency>
   <groupId>io.camunda</groupId>
-  <artifactId>zeebe-process-test-extension</artifactId>
+  <artifactId>zeebe-process-test-extension-testcontainer</artifactId>
   <version>X.Y.Z</version>
   <scope>test</scope>
 </dependency>
 ```
 
-**Note**: This snapshot version is bound to change in the future. Only use this when you want to play around with the project.
+#### Embedded (JDK 17+)
+
+If you are building your project with JDK 17+ you can make use of an embedded Zeebe test engine. The
+advantage of using this instead of the testcontainer version is that this is the faster solution.
+This also does not require Docker to be running. There is also a downside to this solution. The JDK
+requirement is bound to the Java version of the Zeebe engine. Whenever this Java version gets updated,
+you'd either have to [switch to the testcontainer version](#switching-between-testcontainers-and-embedded),
+or upgrade your own JDK to match the Zeebe engine.
+
+```xml
+<dependency>
+  <groupId>io.camunda</groupId>
+  <artifactId>zeebe-process-test-extension</artifactId>
+  <version>X.Y.Z</version>
+  <scope>test</scope>
+</dependency>
+```
 
 ### Annotation
 
 Annotate your test class with the `@ZeebeProcessTest` annotation. This annotation will do a couple of things:
 
-1. It will create and start the in memory engine. This will be a new engine for each test case.
+1. It will manage the [lifecycle](#engine-lifecycle) of the testcontainer / embedded test engine
 2. It will create a client which can be used to interact with the engine.
 3. It will (optionally) inject 3 fields in your test class:
    1. `ZeebeTestEngine` - This is the engine that will run your process. It will provide some basic functionality
       to help you write your tests, such as waiting for an idle state and increasing the time.
-   2. `ZeebeClient` - This is the client that allows you to communicate with the engine.
-      It allows you to send commands to the engine.
+   2. `ZeebeClient` - This is the client that allows you to  send commands to the engine, such as
+      starting a process instance.
```

Comment:
:wrench: Is this obvious or should we state it explicitly?
_The interface of this client is identical to the interface you use to connect to a real Zeebe engine._